src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -679,13 +679,22 @@ public function stripParentheses($expression)
 
     /**
      * Register a custom Blade compiler.
-     *
-     * @param  callable  $compiler
-     * @return void
      */
-    public function extend(callable $compiler)
+    public function extend(callable $compiler, ?string $name = null): void
+    {
+        if (! empty($name)) {
```

Comment:
How about `if (filled($name)) {`?

`filled()` will trim the string before checking if it is not blank. And also check for non-null values.

Reply:
Thanks! I'll update if there's interest in merging the PR.

Reply:
Updated.